File: app/assets/javascripts/desk/store/Checklist.js
Ext.define('Desk.store.Checklist', {
  extend       : 'Ext.data.Store',
  alias        : 'store.desk-Checklist',
  model        : 'Desk.model.Checklist',
  autoLoad     : false,
  storeId      : 'ChecklistStore',
  remoteFilter : true,
  remoteSort   : true,

  constructor  : function (config) {
    var me    = this;
    me.callParent(this);
    
    var proxy = Ext.Object.merge({}, me.getProxy());
    me.setProxy(proxy);
  } // constructor

}); // Ext.define('Desk.store.Checklist'

File: app/services/desk/checklist.rb
class Desk::Checklist::Service
  include Buildit::Service::Base

  service 'Checklist', 'Desk.service'

  connected_mode(Buildit::Service::Backend::Crud) do |config|
    config.model = Desk::Checklist
  end

end # class Desk::Checklist::Service
